### PR TITLE
Bigquery acceptance tests

### PR DESCRIPTION
**Pull request by maryam21**
## What
Acceptance tests for source Bigquery connector

## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-integrations/connectors/source-bigquery/acceptance-test-config.yml [hunk 1]
```diff
@@ -9,8 +10,8 @@ acceptance_tests:
     tests:
       - config_path: "secrets/config.json"
         status: "succeed"
-      - config_path: "integration_tests/invalid_config.json"
-        status: "failed"
+      # - config_path: "integration_tests/invalid_config.json"
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-bigquery/acceptance-test-config.yml:13]: Is there a reason we disable the `failed` config checks?

File: airbyte-integrations/connectors/source-bigquery/acceptance-test-config.yml [hunk 2]
```diff
@@ -20,9 +21,9 @@ acceptance_tests:
         configured_catalog_path: "integration_tests/configured_catalog.json"
         empty_streams: []
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-bigquery/acceptance-test-config.yml:22]: If there are no `empty_streams` - we shouldn't specify this.

File: airbyte-integrations/connectors/source-bigquery/acceptance-test-config.yml [hunk 3]
```diff
@@ -20,9 +21,9 @@ acceptance_tests:
...
 # TODO uncomment this block to specify that the tests should assert the connector outputs the records provided in the input file a file
-#        expect_records:
-#          path: "integration_tests/expected_records.jsonl"
-#          exact_order: no
+      #  expect_records:
+      #    path: "integration_tests/expected_records.jsonl"
+      #    exact_order: no
   incremental: 
     bypass_reason: "This connector does not implement incremental sync"
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-bigquery/acceptance-test-config.yml:28]: Could you please clarify, why we don't test the `incremental` functionality here? Is it intentional for now, basically?

File: airbyte-integrations/connectors/source-bigquery/integration_tests/invalid_config.json [hunk 4]
```diff
@@ -1,3 +1,8 @@
 {
-  "todo-wrong-field": "this should be an incomplete config file, used in standard tests"
-}
+  "dataset_id": "test.test_airbyte_dataset",
+  "project_id": "test",
+  "credentials_json": "{\"type\": \"service_account\",\"project_id\": \"test\",\"private_key_id\": \"1\",\"private_key\": \"-----BEGIN PRIVATE KEY-----\n-----END PRIVATE KEY-----\n\",\"client_email\": \"test@test.iam.gserviceaccount.com\",\"client_id\": \"1\",\"auth_uri\": \"https://accounts.google.com/o/oauth2/auth\",\"token_uri\": \"https://oauth2.googleapis.com/token\",\"auth_provider_x509_cert_url\": \"https://www.googleapis.com/oauth2/v1/certs\",\"client_x509_cert_url\": \"\",\"universe_domain\": \"googleapis.com\"}",
+  "streams": [],
+  "replication_method": {"method": "History"},
+  "slice_range": "-1"
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-bigquery/integration_tests/invalid_config.json:7]: Does this is the indicator the source should look at the data -1 day from the starting point? If so, can we have the `lookback window` instead, having the `slice_range` always a positive int value?

File: airbyte-integrations/connectors/source-bigquery/metadata.yaml [hunk 5]
```diff
@@ -22,13 +23,14 @@ data:
   dockerImageTag: 0.1.0
   dockerRepository: airbyte/source-bigquery
   githubIssueLabel: source-bigquery
-  icon: bigquery.svg
+  icon: icon.svg
   license: MIT
   name: Bigquery
-  releaseDate: TODO
+  releaseDate: 2024-06-17
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-bigquery/metadata.yaml:29]: Why do we need this `releaseDate`?

File: airbyte-integrations/connectors/source-bigquery/integration_tests/configured_catalog.json [hunk 6]
```diff
@@ -2,21 +2,18 @@
...
-      "stream": {
-        "name": "employees",
-        "json_schema": {},
-        "supported_sync_modes": ["full_refresh", "incremental"]
-      },
-      "sync_mode": "incremental",
-      "destination_sync_mode": "append"
+      "cursor_field": null,
+      "destination_sync_mode": "append",
+      "primary_key": null
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-bigquery/integration_tests/configured_catalog.json:16]: If the `"source_defined_primary_key"` is defined as `[["id"]]`, the `primary_key` should reflect the same value.

File: airbyte-integrations/connectors/source-bigquery/acceptance-test-config.yml (hunk 3 above)
Comment (maryam21) [reply airbyte-integrations/connectors/source-bigquery/acceptance-test-config.yml]: @bazarnov 
I was originally planning to do it in other integration tests that's why. Now I updated the acceptance tests to also include the incremental here https://github.com/artefactory/airbyte/pull/28/commits/26d29a27deb987ac1d86ec7aba9478d4c89dd148

File: airbyte-integrations/connectors/source-bigquery/source_bigquery/source.py [hunk 7]
```diff
@@ -71,6 +72,18 @@ def __init__(self, catalog: Optional[ConfiguredAirbyteCatalog], config: Optional
         self.catalog = catalog
         self.state = state
 
+    @staticmethod
+    def validate_config(config: MutableMapping[str, Any]) -> MutableMapping[str, Any]:
+        slice_range = float(config.get("slice_range", _DEFAULT_SLICE_RANGE))
```
[the remark this answers is not in the document]
Comment (maryam21) [reply airbyte-integrations/connectors/source-bigquery/source_bigquery/source.py:77]: I tried doing this, and when I did the form validation fails as the numbers entered are not considered ints

File: airbyte-integrations/connectors/source-bigquery/source_bigquery/source.py [hunk 8]
```diff
@@ -71,6 +72,18 @@ def __init__(self, catalog: Optional[ConfiguredAirbyteCatalog], config: Optional
...
+        if slice_range and slice_range <= 0:
+            message = f"Invalid slice range {slice_range}. Please use only positive integer values."
```
[the remark this answers is not in the document]
Comment (maryam21) [reply airbyte-integrations/connectors/source-bigquery/source_bigquery/source.py:79]: Ok, I did not know this, I will update the spec

File: airbyte-integrations/connectors/source-bigquery/integration_tests/configured_catalog.json (hunk 6 above)
Comment (maryam21) [reply airbyte-integrations/connectors/source-bigquery/integration_tests/configured_catalog.json]: fixed

File: airbyte-integrations/connectors/source-bigquery/metadata.yaml (hunk 5 above)
Comment (maryam21) [reply airbyte-integrations/connectors/source-bigquery/metadata.yaml:29]: @bazarnov 
If I do not fill it the spec acceptance test fails

File: airbyte-integrations/connectors/source-bigquery/integration_tests/invalid_config.json (hunk 4 above)
Comment (maryam21) [reply airbyte-integrations/connectors/source-bigquery/integration_tests/invalid_config.json:7]: No this is just to test if check connection fails on an invalid value of `slice_range`

File: airbyte-integrations/connectors/source-bigquery/source_bigquery/source.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (bazarnov) [reply airbyte-integrations/connectors/source-bigquery/source_bigquery/source.py:77]: I'm fine with this additional casting, but having something like this:
```yaml
...
slice_range:
  type: number
  description: The stream partitions range in minutes
  title: Slice Range
  default: 525600.0
  minimum: 0.0
  maximum: 525600.0
...
```
It should work, I believe. I'm not blocking on this, but I'm super curious as to why this does not work for you.

File: airbyte-integrations/connectors/source-bigquery/source_bigquery/source.py [hunk 9]
```diff
@@ -110,7 +125,10 @@ def check_connection(self, logger, config: Mapping[str, Any]) -> Tuple[bool, any
                         failure_type=FailureType.transient_error,
                         message=error_msg,
                     )
-            return True, error_msg
+            return False, error_msg
+        except Exception as error:
```
[the remark this answers is not in the document]
Comment (bazarnov) [reply airbyte-integrations/connectors/source-bigquery/source_bigquery/source.py:129]: If there is something like the `FailureType.system_error`, based on the fact this `Exception` is not expected, but we ensure we could catch some - it's also fine to have raised a `FailureType.system_error` instead.

**Review by bazarnov — APPROVED** [review]
Looks great, few minor comments left, consider them non-blocking.

File: airbyte-integrations/connectors/source-bigquery/integration_tests/resources/expected_spec.json [hunk 10]
```diff
@@ -0,0 +1,30 @@
+{
+  "documentationUrl": "https://docs.airbyte.com/integrations/sources/bigquery",
+  "supportsIncremental": true,
+  "supported_destination_sync_modes": [],
+  "supported_sync_modes": ["overwrite", "append", "append_dedup"],
```
[the remark this answers is not in the document]
Comment (maryam21) [reply airbyte-integrations/connectors/source-bigquery/integration_tests/resources/expected_spec.json]: done here https://github.com/artefactory/airbyte/pull/28/commits/9819092406d1310a651cf0436e4623eeac17ecd6

File: airbyte-integrations/connectors/source-bigquery/acceptance-test-config.yml (hunk 2 above)
Comment (maryam21) [reply airbyte-integrations/connectors/source-bigquery/acceptance-test-config.yml]: done here https://github.com/artefactory/airbyte/pull/28/commits/70f624cde28a92c777bd6db4515f36b0fbc8c908

**Pull request merged**